src/Core/Services/MetadataProviders/MsSqlMetadataProvider.cs

```diff
@@ -80,5 +85,156 @@ public override async Task PopulateTriggerMetadataForTable(string entityName, st
                 }
             }
         }
+
+        /// <inheritdoc/>
+        protected override void PopulateColumnDefinitionWithHasDefaultAndDbType(
+            SourceDefinition sourceDefinition,
+            DataTable allColumnsInTable)
+        {
+            foreach (DataRow columnInfo in allColumnsInTable.Rows)
+            {
+                string columnName = (string)columnInfo["COLUMN_NAME"];
+                bool hasDefault =
+                    Type.GetTypeCode(columnInfo["COLUMN_DEFAULT"].GetType()) != TypeCode.DBNull;
+                if (sourceDefinition.Columns.TryGetValue(columnName, out ColumnDefinition? columnDefinition))
+                {
+                    columnDefinition.HasDefault = hasDefault;
+
+                    if (hasDefault)
+                    {
+                        columnDefinition.DefaultValue = columnInfo["COLUMN_DEFAULT"];
+                    }
+
+                    columnDefinition.DbType = TypeHelper.GetDbTypeFromSystemType(columnDefinition.SystemType);
+                    if (columnDefinition.SystemType == typeof(DateTime) || columnDefinition.SystemType == typeof(DateTimeOffset))
+                    {
+                        // MsSql types like date,smalldatetime,datetime,datetime2 are mapped to the same .NET type of DateTime.
+                        // Thus to determine the actual dbtype, we use the underlying MsSql type instead of the .NET type.
+                        DbType dbType;
+                        string sqlType = (string)columnInfo["DATA_TYPE"];
+                        if (TryResolveDbType(sqlType, out dbType))
+                        {
+                            columnDefinition.DbType = dbType;
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override async Task FillSchemaForStoredProcedureAsync(
+            Entity procedureEntity,
+            string entityName,
+            string schemaName,
+            string storedProcedureSourceName,
+            StoredProcedureDefinition storedProcedureDefinition)
+        {
+            using DbConnection conn = new SqlConnection();
```

Comment:
this code is quite similar to the code in the base class for this same function, is there anyway we can have this in a form factor so that the similar code can be shared?

Reply:
Even though the code is similar, we want to not have mssql specific logic in the base class code. Also, if I move back the code to the base class, we would need to have conditional for datatime types and then call the overridden method for populating DbType. For database types other than MsSql, calling the overridden method again and again won't make sense.

Reply:
I dont think this refactor needs to combine MsSql specific code, will follow up in the issue created.